src/controllers/csi/provisioner/agent.go

```diff
@@ -91,25 +112,25 @@ func setupImageInstaller(ctx context.Context, fs afero.Fs, apiReader client.Read
 	}
 	imageInstaller := image.NewImageInstaller(fs, &image.Properties{
 		ImageUri:     dynakube.CodeModulesImage(),
+		ImageDigest:  digest,
+		PathResolver: pathResolver,
 		DockerConfig: *dockerConfig})
 	return imageInstaller, nil
 }
 
-func (updater *agentUpdater) updateAgent(installedVersion, tenantUUID string, latestProcessModuleConfigCache *processModuleConfigCache) (string, error) {
+func (updater *agentUpdater) updateAgent(installedVersion string, latestProcessModuleConfigCache *processModuleConfigCache) (string, error) {
 	defer updater.cleanCertsIfPresent()
 	var updatedVersion string
-	if updater.VersionDirNotPresent() || installedVersion == "" {
+	if updater.versionDirNotPresent() || installedVersion == "" {
 		log.Info("updating agent",
 			"target version", updater.targetVersion,
 			"installed version", installedVersion,
 			"target directory", updater.targetDir)
-		_ = updater.fs.MkdirAll(updater.targetDir, 0755)
-
 		if err := updater.installer.InstallAgent(updater.targetDir); err != nil {
```

Comment:
I get that it's not part of your changes, but I just want to mention that the function is called `updateAgent`, but it also installs it. Then, further down, it updates the agents config, which is not really the agent itself. I'd argue that this function wants to be split up as it does two different things none of which really do what the functions name suggests

Reply:
so the agent updater as a whole needs to be refactored, we are planning to give it the webhook rework treatment that is why I didn't try to refactor the whole thing right now